src/main/java/org/jitsi/jicofo/recording/jibri/JibriRecorder.java

```diff
@@ -323,33 +340,11 @@ private void processJibriIqFromMeet(JibriIq iq, XmppChatMember sender)
             if (jibriJid == null)
             {
                 sendErrorResponse(
-                    iq, XMPPError.Condition.service_unavailable, null);
+                        iq, XMPPError.Condition.service_unavailable, null);
                 return;
             }
 
-            final String roomName = MucUtil.extractName(senderMucJid);
-
-            try
-            {
-                final XMPPError err = startJibri(
-                        jibriJid, roomName, iq.getStreamId());
-                if (err == null)
-                {
-                    // ACK the original request
-                    sendResultResponse(iq);
-                }
-                else
-                {
-                    sendErrorResponse(
-                            iq, XMPPError.Condition.interna_server_error, null);
-                }
-            }
-            catch (final SmackException.NoResponseException e)
-            {
-                sendErrorResponse(
-                        iq, XMPPError.Condition.request_timeout, null);
-                return;
-            }
+            scheduleRetryStartJibri(jibriJid, iq);
```

Comment:
In this case returns combined with multiple "else if" structure make it easy to forget about "return" in the right place.

I'm not aware of method that will "insert a stack frame above the current one". Maybe wrapping the thing in a parent method or using finally block, but not sure if I understand what you meant correctly.